src/CozyUtils.js

```diff
@@ -153,30 +153,61 @@ class CozyUtils {
    * @param  {array} files   files received from the toutatice API call
    * @returns {object}
    */
-  computeShortcuts(files) {
+  async computeShortcuts(files) {
     log('info', 'computeShortcuts starts')
     let schoolShortcuts = []
     let favShortcuts = []
+    let processedIcons = 0
+    let unprocessedIcons = 0
     for (const file of files) {
       if (file.hubMetadata.favori) {
-        favShortcuts.push({
+        const appToSave = {
           ...file,
           vendorRef: file.hubMetadata.idInterne,
           filename: `${file.title}.url`,
           filestream: `[InternetShortcut]\nURL=${file.url}`,
-          shouldReplaceFile: true
-        })
+          shouldReplaceFile: true,
+          fileAttributes: {
+            metadata: {}
+          }
+        }
+        if (appToSave.icon && appToSave.icon.match(/https?:\/\//g)) {
+          appToSave.fileAttributes.metadata.icon = await this.fetchIcon(
+            appToSave.icon
+          )
+          delete appToSave.icon
+          processedIcons++
+        } else if (!appToSave.icon || !appToSave.icon.match(/https?:\/\//g)) {
+          unprocessedIcons++
+        }
+        favShortcuts.push(appToSave)
       } else {
         const appToSave = {
           ...file,
           vendorRef: file.hubMetadata.idInterne,
           filename: `${file.title}.url`,
           filestream: `[InternetShortcut]\nURL=${file.url}`,
-          shouldReplaceFile: true
+          shouldReplaceFile: true,
+          fileAttributes: {
+            metadata: {}
+          }
```

Comment:
Pourquoi avoir ajouté cet objet vide à chaque fois et non pas que si y a un match sur l'icon ?

Reply:
Parce que quand j'ai essayé de faire ça, l'assignation de `fileAttributes.metadata.icon` me renvoyais une erreur "icon of undefined", j'en ai déduis qu'il fallait que `metadata` soit présent avant de vouloir faire la-dite assignation, donc je la rajoute dans tout les cas